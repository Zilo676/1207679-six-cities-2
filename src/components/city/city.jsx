import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {ActionCreator} from '../../reducer/city/city';

const City = (props) => {
  const {city, onClick, onActiveItem} = props;

  return (<li className="locations__item" onClick={() => {
    onActiveItem(city);
    onClick(city);
  }}>
    <a className="locations__item-link tabs__item" href="#">
      <span>{city}</span>
    </a>
  </li>);
};

City.propTypes = {
  city: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  onActiveItem: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onClick: (city) => {
    dispatch(ActionCreator.setCity(city));
  },
});

export {City};

export default connect(null, mapDispatchToProps)(City);
