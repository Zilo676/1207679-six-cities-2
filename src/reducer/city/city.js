const CITIES = [`Amsterdam`, `Brussels`, `Cologne`, `Dusseldorf`, `Hamburg`, `Paris`];

const initialState = {
  cities: CITIES,
  city: `Amsterdam`,
};

const ActionType = {
  SET_CITY: `SET_CITY`,
};

const ActionCreator = {
  setCity: (city) => {
    return {
      type: ActionType.SET_CITY,
      payload: city,
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_CITY:
      return Object.assign({}, state, {
        city: action.payload,
      });
  }

  return state;
};

export {
  ActionCreator,
  ActionType,
  reducer,
};
