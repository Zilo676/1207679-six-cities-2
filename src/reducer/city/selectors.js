import NameSpace from '../name-spaces';

const NAME_SPACE = NameSpace.CITY;

const getCurrentCity = (state) => {
  return state[NAME_SPACE].city;
};

const getAllCities = (state) => {
  return state[NAME_SPACE].cities;
};

export {
  getAllCities,
  getCurrentCity,
};
