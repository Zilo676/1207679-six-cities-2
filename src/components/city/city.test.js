import React from 'react';
import renderer from 'react-test-renderer';

import {City} from './city.jsx';

it(`Correctly rendered`, () => {
  const mockFunc = jest.fn();
  const tree = renderer.create(<City
    city={`Amsterdam`}
    onClick={mockFunc}
    onActiveItem={mockFunc}
    currentCity={`ASDB`}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
