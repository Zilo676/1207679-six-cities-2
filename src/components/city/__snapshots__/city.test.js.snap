// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendered 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <span
    className="locations__item-link tabs__item "
  >
    Amsterdam
  </span>
</li>
`;
